cs.c

```diff
@@ -72,196 +72,240 @@
+			| CS_MODE_MIPS32R6 | CS_MODE_BIG_ENDIAN | CS_MODE_MIPS2 | CS_MODE_MIPS3), \
+	}
+#define CS_ARCH_CONFIG_X86 \
+	{ \
+		X86_global_init, \
+		X86_option, \
+		~(CS_MODE_LITTLE_ENDIAN | CS_MODE_32 | CS_MODE_64 | CS_MODE_16), \
+	}
+#define CS_ARCH_CONFIG_PPC \
+	{ \
+		PPC_global_init, \
+		PPC_option, \
+		~(CS_MODE_LITTLE_ENDIAN | CS_MODE_32 | CS_MODE_64 | CS_MODE_BIG_ENDIAN \
+				| CS_MODE_QPX | CS_MODE_PS | CS_MODE_BOOKE), \
+	}
+#define CS_ARCH_CONFIG_SPARC \
+	{ \
+		Sparc_global_init, \
+		Sparc_option, \
+		~(CS_MODE_BIG_ENDIAN | CS_MODE_V9), \
+	}
+#define CS_ARCH_CONFIG_SYSZ \
+	{ \
+		SystemZ_global_init, \
+		SystemZ_option, \
+		~(CS_MODE_BIG_ENDIAN), \
+	}
+#define CS_ARCH_CONFIG_XCORE \
+	{ \
+		XCore_global_init, \
+		XCore_option, \
+		~(CS_MODE_BIG_ENDIAN), \
+	}
+#define CS_ARCH_CONFIG_M68K \
+	{ \
+		M68K_global_init, \
+		M68K_option, \
+		~(CS_MODE_BIG_ENDIAN | CS_MODE_M68K_000 | CS_MODE_M68K_010 | CS_MODE_M68K_020 \
+				| CS_MODE_M68K_030 | CS_MODE_M68K_040 | CS_MODE_M68K_060), \
+	}
+#define CS_ARCH_CONFIG_TMS320C64X \
+	{ \
+		TMS320C64x_global_init, \
+		TMS320C64x_option, \
+		~(CS_MODE_BIG_ENDIAN), \
+	}
+#define CS_ARCH_CONFIG_M680X \
+	{ \
+		M680X_global_init, \
+		M680X_option, \
+		~(CS_MODE_M680X_6301 | CS_MODE_M680X_6309 | CS_MODE_M680X_6800 \
+				| CS_MODE_M680X_6801 | CS_MODE_M680X_6805 | CS_MODE_M680X_6808 \
+				| CS_MODE_M680X_6809 | CS_MODE_M680X_6811 | CS_MODE_M680X_CPU12 \
+				| CS_MODE_M680X_HCS08), \
+	}
+#define CS_ARCH_CONFIG_EVM \
+	{ \
+		EVM_global_init, \
+		EVM_option, \
+		0, \
+	}
+#define CS_ARCH_CONFIG_MOS65XX \
+	{ \
+		MOS65XX_global_init, \
+		MOS65XX_option, \
+		~(CS_MODE_LITTLE_ENDIAN | CS_MODE_MOS65XX_6502 | CS_MODE_MOS65XX_65C02 \
+				| CS_MODE_MOS65XX_W65C02 | CS_MODE_MOS65XX_65816_LONG_MX), \
+	}
+#define CS_ARCH_CONFIG_WASM \
+	{ \
+		WASM_global_init, \
+		WASM_option, \
+		0, \
+	}
+#define CS_ARCH_CONFIG_BPF \
+	{ \
+		BPF_global_init, \
+		BPF_option, \
+		~(CS_MODE_LITTLE_ENDIAN | CS_MODE_BPF_CLASSIC | CS_MODE_BPF_EXTENDED \
+				| CS_MODE_BIG_ENDIAN), \
+	}
+#define CS_ARCH_CONFIG_RISCV \
+	{ \
+		RISCV_global_init, \
+		RISCV_option, \
+		~(CS_MODE_RISCV32 | CS_MODE_RISCV64 | CS_MODE_RISCVC), \
+	}
+#define CS_ARCH_CONFIG_SH \
+	{ \
+		SH_global_init, \
+		SH_option, \
+		~(CS_MODE_SH2 | CS_MODE_SH2A | CS_MODE_SH3 | \
+		  CS_MODE_SH4 | CS_MODE_SH4A | \
+		  CS_MODE_SHFPU | CS_MODE_SHDSP|CS_MODE_BIG_ENDIAN), \
+	}
+#define CS_ARCH_CONFIG_TRICORE \
+	{ \
+		TRICORE_global_init, \
+		TRICORE_option, \
+		~(CS_MODE_TRICORE_110 | CS_MODE_TRICORE_120 | CS_MODE_TRICORE_130 \
+		| CS_MODE_TRICORE_131 | CS_MODE_TRICORE_160 | CS_MODE_TRICORE_161 \
+		| CS_MODE_TRICORE_162 | CS_MODE_LITTLE_ENDIAN), \
+	}
+#define CS_ARCH_CONFIG_ALPHA \
+	{ \
+		ALPHA_global_init, \
+		ALPHA_option, \
+		~(CS_MODE_LITTLE_ENDIAN | CS_MODE_BIG_ENDIAN), \
+	}
+
+#ifdef CAPSTONE_USE_ARCH_REGISTRATION
+static cs_arch_config arch_configs[MAX_ARCH];
+static uint32_t all_arch;
```

Comment:
In `cs.c:1066` there is a NULL check missing before `arch_option` is called. Please add this as well.